lib/memstore.ts

```diff
@@ -48,27 +17,66 @@ export type MemoryCookieStoreIndex = {
   }
 }
 
+/**
+ * An in-memory {@link Store} implementation for {@link CookieJar}. This is the default implementation used by
+ * {@link CookieJar} and supports both async and sync operations. Also supports serialization, getAllCookies, and removeAllCookies.
+ * @public
+ */
 export class MemoryCookieStore extends Store {
+  /**
+   * This value is `true` since {@link MemoryCookieStore} implements synchronous functionality.
+   */
   override synchronous: boolean
+
+  /**
+   * @internal
+   */
   idx: MemoryCookieStoreIndex
 
+  /**
+   * Create a new {@link MemoryCookieStore}.
+   */
   constructor() {
     super()
     this.synchronous = true
     this.idx = Object.create(null) as MemoryCookieStoreIndex
   }
 
+  /**
+   * Retrieve a {@link Cookie} with the given `domain`, `path`, and `key` (`name`). The RFC maintains that exactly
+   * one of these cookies should exist in a store. If the store is using versioning, this means that the latest or
+   * newest such cookie should be returned.
+   *
+   * Callback takes an error and the resulting Cookie object. If no cookie is found then null MUST be passed instead (that is, not an error).
```

Comment:
```suggestion
```
No callback in this signature!

![image](https://github.com/salesforce/tough-cookie/assets/62956339/97a312c3-a4aa-4766-868f-935a408ffa41)